src/transport/heat_model.hh

```diff
@@ -234,13 +234,32 @@ public:
 	}
 
     /// Returns number of transported substances.
-    inline unsigned int n_substances()
+    inline unsigned int n_substances() override
     { return 1; }
 
     /// Returns reference to the vector of substance names.
     inline SubstanceList &substances()
     { return substances_; }
 
+    const vector<unsigned int> &get_subst_idx() override
+	{ return subst_idx; }
+
+    std::shared_ptr<Balance> balance() const override {
+        return this->balance_;
+    }
+
+    std::shared_ptr<FieldFE<3, FieldValue<3>::VectorFixed>> velocity_field_ptr() const override {
```

Comment:
Try to get rid of virtual getters, some are already implemented in parent classes, others can be done in similar way.